schism/compiler.ss

```diff
@@ -210,33 +230,47 @@
        (define (char-ci<? c1 c2)
          (< (char->integer c1) (char->integer c2)))
        (define (list->string ls)
-         (if (pair? ls)
-             ;; For now we represent strings as lists of characters. That
-             ;; means converting between the two is just a matter of
-             ;; changing the tags.
-             (%set-tag ls ,(string-tag))
-             (error 'list->string "list->string: not a pair")))
+         (list->string-inner (make-string (length ls)) ls 0))
```

Comment:
Given that there's not the nice retag property for list->string / string->list, perhaps consider removing list->string and string->list and adapting callers.

Reply:
`list->string` and `string->list` are still part of Scheme, so we might as well keep it around. I agree though that we should make sure the callers aren't still assuming this is a cheap operation though.